azure-devops/azext_devops/dev/team/wiki.py

```diff
@@ -0,0 +1,220 @@
+from knack.util import CLIError
+from azext_devops.dev.common.services import (get_wiki_client,
+                                              get_core_client,
+                                              get_git_client,
+                                              resolve_instance,
+                                              resolve_instance_and_project,
+                                              resolve_instance_project_and_repo)
+
+
+_DEFAULT_PAGE_ADD_MESSAGE = 'Added a new page using Azure DevOps CLI'
+_DEFAULT_PAGE_UPDATE_MESSAGE = 'Updated the page using Azure DevOps CLI'
+_DEFAULT_PAGE_DELETE_MESSAGE = 'Deleted the page using Azure DevOps CLI'
+
+
+def create_wiki(name, wiki_type='projectwiki', mapped_path=None, version=None,
+                organization=None, project=None, repository=None, detect=None):
+    """Create a wiki.
+    :param name: Name of the new wiki.
+    :type name: str
+    :param wiki_type: Type of wiki to create.
+    :type wiki_type: str
+    :param version: Repository branch name to publish the code wiki from. Only required for codewiki type.
+    :type version: str
+    :param mapped_path: Mapped path of the new wiki e.g. '/' to publish from root of repository.
+    Not required for project wiki.
+    :type mapped_path: str
+    :param repository: Name or ID of the repository to publish the wiki from. Only required for codewiki type.
+    :type repository: str
+    """
+    organization, project, repository = resolve_instance_project_and_repo(detect=detect,
+                                                                          organization=organization,
+                                                                          project=project,
+                                                                          repo=repository)
+    wiki_client = get_wiki_client(organization)
+    from azext_devops.devops_sdk.v5_0.wiki.models import WikiCreateParametersV2
+    wiki_params = WikiCreateParametersV2()
+    wiki_params.name = name
+    wiki_params.type = wiki_type
+    project_id = _get_project_id_from_name(organization=organization,
+                                           project=project)
+    wiki_params.project_id = project_id
+    repository_id = _get_repository_id_from_name(organization=organization,
+                                                 project=project,
+                                                 repository=repository)
+    wiki_params.repository_id = repository_id
+    if mapped_path:
+        wiki_params.mapped_path = mapped_path
+    if version:
+        from azext_devops.devops_sdk.v5_0.wiki.models import GitVersionDescriptor
+        version_descriptor = GitVersionDescriptor()
+        version_descriptor.version = version
+        wiki_params.version = version_descriptor
+    return wiki_client.create_wiki(wiki_create_params=wiki_params, project=project)
+
+
+def delete_wiki(wiki, organization=None, project=None, detect=None):  # pylint: disable=redefined-builtin
```

Comment:
what is this needed for?
`# pylint: disable=redefined-builtin`

and in some other places as well

Reply:
This one is redundant. At other places it is for "open".